src/cose/util.rs

```diff
@@ -94,42 +110,34 @@ pub fn get_sig_struct_bytes(
 //  ]
 pub fn build_cose_signature(
     cert_chain: &[&[u8]],
-    ee_cert: &[u8],
-    sig_bytes: &[u8],
-    alg: &CoseSignatureType,
+    ee_certs: &[&[u8]],
+    sig_bytes: &Vec<Vec<u8>>,
+    alg: &[SignatureAlgorithm],
```

Comment:
Similar idea here - I think we should have a vec of a new type that encapsulates these values.